Services/GEV/Desktop/classes/class.gevMyTrainingsApTableGUI.php

```diff
@@ -132,7 +118,86 @@ protected function fillRow($a_set) {
+		$this->tpl->setVariable("TARGET_GROUP_DESC", $a_set["target_group_desc"]);
+		$this->tpl->setVariable("GOALS", $a_set["goals"]);
+		$this->tpl->setVariable("CONTENTS", $a_set["content"]);
+		$this->tpl->setVariable("MBMRLST_LINK", $memberlist_link);
+		$this->tpl->setVariable("MBMRLST_LINK_TXT", $this->gLng->txt('gev_mytrainingsap_btn_memberlist'));
+		$this->tpl->setVariable("SIGNATURE_LIST_LINK", $signature_list_link);
+		$this->tpl->setVariable("SIGNATURE_LIST_LINK_TXT", $this->gLng->txt('gev_signature_list'));
+		if ($a_set['may_finalize']) {
+			$this->tpl->setCurrentBlock("set_stat");
+			$this->tpl->setVariable("SETSTAT_LINK", $setstatus_link);
+			$this->tpl->setVariable("SETSTAT_LINK_TXT", $this->gLng->txt('gev_mytrainingsap_btn_setstatus'));
+			$this->tpl->parseCurrentBlock();
+		}
+		if( $crs_utils->canViewBookings($this->user_id) ) {
+			$this->tpl->setCurrentBlock("view_bookings");
+			$this->tpl->setVariable("VIEW_BOOKINGS_LINK", $bookings_link);
+			$this->tpl->setVariable("VIEW_BOOKINGS_LINK_TXT", $this->gLng->txt('gev_mytrainingsap_btn_bookings'));
+			$this->tpl->parseCurrentBlock();
+		}
+
+		$actions = "";
+		if($crs_utils->isVirtualTraining()) {
+			$this->tpl->setVariable("VC_HEADER", "Zugangsdaten virtueller Klassenraum");
+		
+			if($crs_utils->getVirtualClassLoginTutor()) {
+			$actions .= "Login: ".$crs_utils->getVirtualClassLoginTutor()."<br />";
+			}
+
+			if($crs_utils->getVirtualClassPasswordTutor()) {
+				$actions .= "Passwort: ".$crs_utils->getVirtualClassPasswordTutor();
+			}
+
+			$this->tpl->setVariable("VC_DATA", $actions);
+		}
+		
+	}
+
+	protected function addActionMenu($a_set) {
+		include_once("Services/UIComponent/AdvancedSelectionList/classes/class.ilAdvancedSelectionListGUI.php");
+		$current_selection_list = new ilAdvancedSelectionListGUI();
+		$current_selection_list->setAsynch(true && false);
+		$current_selection_list->setAsynchUrl(true);
+		$current_selection_list->setListTitle($this->gLng ->txt("actions"));
+		$current_selection_list->setId("act_".$this->getUniqueItemId());
+		$current_selection_list->setSelectionHeaderClass("small");
+		$current_selection_list->setItemLinkClass("xsmall");
+		$current_selection_list->setLinksMode("il_ContainerItemCommand2");
+		$current_selection_list->setHeaderIcon(ilAdvancedSelectionListGUI::DOWN_ARROW_DARK);
+		$current_selection_list->setUseImages(false);
+		$current_selection_list->setAdditionalToggleElement($this->getUniqueItemId("sd"), "ilContainerListItemOuterHighlight");
 		
+		$this->addActionMenuItems($current_selection_list, $a_set);
+
+		return $current_selection_list->getHTML();
+	}
+
+	protected function addActionMenuItems($current_selection_list, $a_set) {
+		foreach ($this->getActionMenuItems($a_set) as $key => $value) {
+			$current_selection_list->addItem($value["title"],"",$value["link"],$value["image"],"",$value["frame"]);
+		}
+	}
+
+	protected function getActionMenuItems($a_set) {
```

Comment:
Vielleicht check ichs nicht richtig, aber gibt es diese Methode nicht schon in der gevMyTrainingsAdminTableGUI?